app/views/ProfileView/index.tsx

```diff
@@ -447,8 +454,75 @@ class ProfileView extends React.Component<IProfileViewProps, IProfileViewState>
 		});
 	};
 
+	deleteOwnAccount = () => {
+		logEvent(events.DELETE_OWN_ACCOUNT);
+		prompt(
+			I18n.t('Are_you_sure_you_want_to_delete_your_account?'),
+			I18n.t('For_your_security_you_must_enter_your_current_password_to_continue'),
+			[
+				{ text: I18n.t('Cancel'), onPress: () => {}, style: 'cancel' },
+				{
+					text: I18n.t('Confirm'),
+					onPress: async (password: string) => {
+						this.setState({ currentPassword: password });
+						const { logoutUser } = this.props;
+						try {
+							await RocketChat.deleteOwnAccount(sha256(password));
+							logoutUser();
+							EventEmitter.emit(LISTENER, { message: I18n.t('User_has_been_deleted') });
+						} catch (error) {
+							// @ts-ignore
+							if (error.data.errorType === 'user-last-owner') {
+								// @ts-ignore
+								const { shouldChangeOwner, shouldBeRemoved } = error.data.details;
+								this.setState({ shouldChangeOwner, shouldBeRemoved, confirmOwnerChangeModalVisible: true });
+							} // @ts-ignore
+							else if (error.data.errorType === 'error-invalid-password') {
+								logEvent(events.DELETE_OWN_ACCOUNT_F);
+								EventEmitter.emit(LISTENER, { message: `Invalid password` });
+							} else {
+								// @ts-ignore
+								EventEmitter.emit(LISTENER, { message: error.data.details });
+								logEvent(events.DELETE_OWN_ACCOUNT_F);
+							}
+						}
+					}
+				}
+			],
+			{
+				type: 'secure-text',
+				cancelable: false
+			}
+		);
+	};
+
+	handleConfirmOwnerChange = async () => {
+		const { currentPassword } = this.state;
+		const { logoutUser } = this.props;
+		try {
+			await RocketChat.deleteOwnAccount(sha256(currentPassword!), true);
+			logoutUser();
```

Comment:
Do we need to logout after deleting account?
Hasn't the account already removed?
<img width="692" alt="image" src="https://user-images.githubusercontent.com/804994/158442936-c06c0dd2-3864-4e66-a977-cd005b0dddb9.png">


If so, we should think of calling functions to only remove local stuff from the app (mmkv, wmdb, etc) and redirect to outside.

Something like we [here](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/lib/methods/logout.ts#L129)
```tsx
await removeServerData({ server });
await removeCurrentServer();
await removeServerDatabase({ server });
```

Reply:
Done 

https://user-images.githubusercontent.com/77742477/158811369-fc58b6c7-ac36-4831-9851-af7cff30afba.mp4

* When there're other logged in servers

https://user-images.githubusercontent.com/77742477/158812073-2b8b4281-465f-4022-bcd8-072313fc6bf3.mp4